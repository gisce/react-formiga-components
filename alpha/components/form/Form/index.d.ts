export * from "./Form";
export * from "./Form.types";
//# sourceMappingURL=index.d.ts.map