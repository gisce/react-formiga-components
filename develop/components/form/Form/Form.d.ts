import { FormProps } from "./Form.types";
export declare const Form: (props: FormProps) => import("react/jsx-runtime").JSX.Element;
//# sourceMappingURL=Form.d.ts.map