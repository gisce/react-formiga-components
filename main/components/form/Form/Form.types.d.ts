/// <reference types="react" />
import { FormInstance } from "antd";
export type FormProps = {
    onFieldsChange?: () => void;
    children?: React.ReactNode;
    initialValues?: any;
    form: FormInstance;
};
//# sourceMappingURL=Form.types.d.ts.map