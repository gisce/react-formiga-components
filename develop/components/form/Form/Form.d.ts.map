{"version":3,"file":"Form.d.ts","sourceRoot":"","sources":["../../../../src/components/form/Form/Form.tsx"],"names":[],"mappings":"AACA,OAAO,EAAE,SAAS,EAAE,MAAM,cAAc,CAAC;AAEzC,eAAO,MAAM,IAAI,UAAW,SAAS,4CAepC,CAAC"}