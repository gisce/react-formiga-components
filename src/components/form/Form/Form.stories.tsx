import { ComponentStory, ComponentMeta } from "@storybook/react";

import {
  CharInput,
  BooleanInput,
  Field,
  Form,
  FileInput,
  ImageInput,
  DateInput,
  TextInput,
  IntegerInput,
  FloatInput,
  SelectionInput,
} from "@/index";
import { useForm } from "antd/es/form/Form";

export default {
  title: "Components/Form/Form",
  component: Form,
  argTypes: {
    children: { table: { disable: true }, control: { disable: true } },
    onFieldsChange: { control: { disable: true } },
  },
} as ComponentMeta<typeof Form>;

const Template: ComponentStory<typeof Form> = (args) => {
  const [antForm] = useForm();
  return <Form {...args} form={antForm} initialValues={{ name: "Hohn" }} />;
};

const formItems = [
  {
    type: CharInput,
    baseProps: {
      required: true,
    },
    fieldProps: {
      name: "name",
      labelOpts: { text: "Name" },
    },
    widgetProps: {},
  },
  {
    type: CharInput,
    baseProps: {
      required: true,
    },
    fieldProps: {
      name: "surnames",
      labelOpts: { text: "Surnames" },
    },
    widgetProps: {},
  },
  {
    type: CharInput,
    baseProps: {
      required: true,
    },
    fieldProps: {
      name: "password",
      labelOpts: { text: "Password" },
    },
    widgetProps: { isPassword: true },
  },
  {
    type: CharInput,
    baseProps: {
      required: true,
    },
    fieldProps: {
      name: "repeat_password",
      labelOpts: { text: "Repeat password" },
    },
    widgetProps: { isPassword: true },
  },
  {
    type: BooleanInput,
    baseProps: {},
    fieldProps: {
      name: "subscribe_to_newsletter",
      labelOpts: { text: "Receive news" },
    },
    widgetProps: {},
  },
  {
    type: FileInput,
    baseProps: {},
    fieldProps: {
      name: "cv",
      labelOpts: { text: "Resume" },
    },
    widgetProps: {},
  },
  {
    type: ImageInput,
    baseProps: {},
    fieldProps: {
      name: "profile_image",
      labelOpts: { text: "Profile image" },
    },
    widgetProps: {},
  },
  {
    type: DateInput,
    baseProps: {},
    fieldProps: {
      name: "birthdate",
      labelOpts: { text: "Birth date" },
    },
    widgetProps: {},
  },
  {
    type: TextInput,
    baseProps: {},
    fieldProps: {
      name: "notes",
      labelOpts: { text: "Additional notes" },
    },
    widgetProps: {},
  },
  {
    type: IntegerInput,
    baseProps: {},
    fieldProps: {
      name: "years_of_experience",
      labelOpts: { text: "Years of experience" },
    },
    widgetProps: {},
  },
  {
    type: FloatInput,
    baseProps: {},
    fieldProps: {
      name: "expected_salary",
      labelOpts: { text: "Expected salary (USD)" },
    },
    widgetProps: { decimalDigits: 2 },
  },
  {
    type: DateInput,
    baseProps: {},
    fieldProps: {
      name: "meeting_date",
      labelOpts: { text: "Approximate meeting timeframe" },
    },
    widgetProps: { showTime: true },
  },
  {
    type: SelectionInput,
    baseProps: {
      required: true,
    },
    fieldProps: {
      name: "role",
      labelOpts: { text: "Role" },
    },
    widgetProps: {
      selectionValues: {
        "1": "Developer",
        "2": "Designer",
        "3": "Manager",
      },
    },
  },
];

function getFormItems() {
  return formItems.map((item) => {
    return FieldWithWidget(item);
  });
}

function FieldWithWidget({
  type: Component,
  baseProps,
  fieldProps,
  widgetProps,
}: any) {
  return (
    <Field key={fieldProps.name} {...{ ...baseProps, ...fieldProps }}>
      <Component {...{ ...baseProps, ...widgetProps }} />
    </Field>
  );
}

export const Basic = Template.bind({
  initialValues: {
    name: "John",
  },
});
Basic.args = {
  children: getFormItems(),
};
