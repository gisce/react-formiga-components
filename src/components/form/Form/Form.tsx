import React from "react";
import { Form as AntForm } from "antd";
import { FormProps } from "./Form.types";

export const Form = (props: FormProps) => {
  const { onFieldsChange, children } = props;
  const [antForm] = AntForm.useForm();

  return (
    <AntForm
      form={antForm}
      onFieldsChange={() => {
        onFieldsChange?.();
      }}
      component={false}
      preserve={false}
    >
      {children}
    </AntForm>
  );
};
