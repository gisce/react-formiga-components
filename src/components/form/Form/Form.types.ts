export type FormProps = {
  onFieldsChange?: () => void;
  children: React.ReactNode;
};
