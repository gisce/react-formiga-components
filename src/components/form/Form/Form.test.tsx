import React from "react";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/testing-react";
import * as stories from "./Form.stories";
import matchMediaMock from "@/mocks/matchMedia.mock";

const { Basic } = composeStories(stories);

describe("Form", () => {
  beforeAll(() => {
    matchMediaMock();
  });

  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName!,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (_storyName, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });

  it("renders basic", () => {
    render(<Basic />);
    expect(true).toBe(true); // TODO: test something
  });
});
