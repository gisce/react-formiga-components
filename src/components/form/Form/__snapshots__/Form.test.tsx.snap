// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Renders Basic story 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 0.25rem;
  padding-top: 0.25rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  padding-right: 0.5rem;
  text-align: right;
}

.c0 {
  padding-bottom: 0.25rem;
  padding-top: 0.25rem;
}

.c1 {
  margin-left: 0.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  margin-bottom: 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c11 .ant-select-selector {
  background-color: rgb(219,234,254) !important;
}

.c6 {
  --tw-bg-opacity: 1 !important;
  background-color: rgba(219,234,254,var(--tw-bg-opacity)) !important;
}

.c6 .ant-input {
  background-color: rgba(219,234,254,var(--tw-bg-opacity)) !important;
}

.c7 {
  --tw-bg-opacity: 1 !important;
  background-color: rgba(219,234,254,var(--tw-bg-opacity)) !important;
}

.c7 .ant-input {
  background-color: rgba(219,234,254,var(--tw-bg-opacity)) !important;
}

.c10 {
  width: 100%;
}

.c9 {
  height: 100% !important;
}

<body>
  <div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Name :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input c6"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Surnames :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input c6"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Password :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input c7"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Repeat password :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input c7"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Receive news :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="c8"
                  data-testid="checkbox_container"
                >
                  <label
                    class="ant-checkbox-wrapper"
                  >
                    <span
                      class="ant-checkbox"
                    >
                      <input
                        class="ant-checkbox-input"
                        data-testid="checkbox_input"
                        id="subscribe_to_newsletter"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Resume :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-row ant-row-no-wrap"
                  style="margin-left: -4px; margin-right: -4px;"
                >
                  <div
                    class="ant-col"
                    style="padding-left: 4px; padding-right: 4px;"
                  >
                    <input
                      id="file"
                      style="display: none;"
                      type="file"
                    />
                    <input
                      class="ant-input ant-input-disabled"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="ant-col"
                    style="padding-left: 4px; padding-right: 4px;"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <button
                          class="ant-btn"
                          type="button"
                        >
                          <span
                            aria-label="folder-open"
                            class="anticon anticon-folder-open"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="folder-open"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M928 444H820V330.4c0-17.7-14.3-32-32-32H473L355.7 186.2a8.15 8.15 0 00-5.5-2.2H96c-17.7 0-32 14.3-32 32v592c0 17.7 14.3 32 32 32h698c13 0 24.8-7.9 29.7-20l134-332c1.5-3.8 2.3-7.9 2.3-12 0-17.7-14.3-32-32-32zM136 256h188.5l119.6 114.4H748V444H238c-13 0-24.8 7.9-29.7 20L136 643.2V256zm635.3 512H159l103.3-256h612.4L771.3 768z"
                              />
                            </svg>
                          </span>
                          <span>
                            Select
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Profile image :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  accept="image/*"
                  id="file"
                  style="display: none;"
                  type="file"
                />
                <div
                  class="ant-space ant-space-vertical"
                >
                  <div
                    class="ant-space-item"
                  >
                    <div
                      class="ant-row ant-row-no-wrap ant-row-center"
                      style="margin-left: 0px; margin-right: -4px;"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                        >
                          <button
                            class="ant-btn ant-btn-icon-only"
                            tooltip="Upload new image"
                            type="button"
                          >
                            <span
                              aria-label="folder-open"
                              class="anticon anticon-folder-open"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="folder-open"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M928 444H820V330.4c0-17.7-14.3-32-32-32H473L355.7 186.2a8.15 8.15 0 00-5.5-2.2H96c-17.7 0-32 14.3-32 32v592c0 17.7 14.3 32 32 32h698c13 0 24.8-7.9 29.7-20l134-332c1.5-3.8 2.3-7.9 2.3-12 0-17.7-14.3-32-32-32zM136 256h188.5l119.6 114.4H748V444H238c-13 0-24.8 7.9-29.7 20L136 643.2V256zm635.3 512H159l103.3-256h612.4L771.3 768z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Birth date :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-picker"
                  style="width: 100%;"
                >
                  <div
                    class="ant-picker-input"
                  >
                    <input
                      autocomplete="off"
                      placeholder="__/__/____"
                      readonly=""
                      size="12"
                      title=""
                      value=""
                    />
                    <span
                      class="ant-picker-suffix"
                    >
                      <span
                        aria-label="calendar"
                        class="anticon anticon-calendar"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="calendar"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Additional notes :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <textarea
                  class="ant-input c9"
                  height="100%"
                  rows="4"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Years of experience :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-input-number c10"
                >
                  <div
                    class="ant-input-number-handler-wrap"
                  >
                    <span
                      aria-disabled="false"
                      aria-label="Increase Value"
                      class="ant-input-number-handler ant-input-number-handler-up"
                      role="button"
                      unselectable="on"
                    >
                      <span
                        aria-label="up"
                        class="anticon anticon-up ant-input-number-handler-up-inner"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="up"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      aria-disabled="false"
                      aria-label="Decrease Value"
                      class="ant-input-number-handler ant-input-number-handler-down"
                      role="button"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-input-number-handler-down-inner"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                  <div
                    class="ant-input-number-input-wrap"
                  >
                    <input
                      aria-valuenow="0"
                      autocomplete="off"
                      class="ant-input-number-input"
                      id="years_of_experience"
                      role="spinbutton"
                      step="1"
                      value="0"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Expected salary (USD) :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-input-number c10"
                >
                  <div
                    class="ant-input-number-handler-wrap"
                  >
                    <span
                      aria-disabled="false"
                      aria-label="Increase Value"
                      class="ant-input-number-handler ant-input-number-handler-up"
                      role="button"
                      unselectable="on"
                    >
                      <span
                        aria-label="up"
                        class="anticon anticon-up ant-input-number-handler-up-inner"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="up"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      aria-disabled="false"
                      aria-label="Decrease Value"
                      class="ant-input-number-handler ant-input-number-handler-down"
                      role="button"
                      unselectable="on"
                    >
                      <span
                        aria-label="down"
                        class="anticon anticon-down ant-input-number-handler-down-inner"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="down"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                  <div
                    class="ant-input-number-input-wrap"
                  >
                    <input
                      aria-valuenow="0"
                      autocomplete="off"
                      class="ant-input-number-input"
                      id="expected_salary"
                      role="spinbutton"
                      step="1"
                      value="0"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Approximate meeting timeframe :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-picker"
                  style="width: 100%;"
                >
                  <div
                    class="ant-picker-input"
                  >
                    <input
                      autocomplete="off"
                      placeholder="__/__/____ __:__:__"
                      readonly=""
                      size="21"
                      title=""
                      value=""
                    />
                    <span
                      class="ant-picker-suffix"
                    >
                      <span
                        aria-label="calendar"
                        class="anticon anticon-calendar"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="calendar"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-middle c0"
    >
      <div
        class="ant-col c1"
        style="flex: 0 0 7rem;"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <span
              class="c4"
            >
              Role :
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
        style="flex-basis: auto;"
      >
        <div
          class="ant-row ant-form-item c5"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-select c11 ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-haspopup="listbox"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title=""
                    />
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
